spec/lib/resque-serializer/serializers/queue_spec.rb

```diff
@@ -21,62 +21,71 @@ def self.perform(*args); end
+  let(:worker_1)   { Resque::Worker.new(queue_name) }
+  let(:worker_2)   { Resque::Worker.new(queue_name) }
 
   before do
-    ResqueSpec.reset!
     Resque.redis.redis.flushall
   end
 
-  describe 'before enqueuing the job' do
-    let(:mutex) { described_class.mutex(args) }
+  subject(:execute_job) { worker_2.reserve }
 
-    subject(:enqueue_job) { Resque.enqueue(described_class, *args) }
-
-    context 'when a lock for the job exists' do
-      before { mutex.lock! }
+  context 'with no jobs in the queue' do
+    before do
+      expect(Resque.size(queue_name)).to eq(0)
+    end
 
-      it 'does not enqueue the job' do
-        expect { enqueue_job }.to_not change {
-          ResqueSpec.queue_for(described_class).size
-        }.from(0)
-      end
+    it 'can enqueue the job' do
+      expect { Resque.enqueue(described_class, *args) }.to change {
+        Resque.size(queue_name)
+      }.from(0).to(1)
+    end
 
-      it 'does not change the mutex' do
-        expect { enqueue_job }.to_not change {
-          mutex.locked?
-        }.from(true)
-      end
+    it 'can not execute any jobs' do
+      expect(execute_job).to be_nil
     end
+  end
 
-    context 'when a lock for the job does not exist' do
-      before { mutex.unlock }
+  context 'with one job in the queue' do
+    before do
+      Resque.enqueue(described_class, *args)
+      expect(Resque.size(queue_name)).to eq(1)
+    end
 
-      it 'enqueues the job' do
-        expect { enqueue_job }.to change {
-          ResqueSpec.queue_for(described_class).size
-        }.from(0).to(1)
-      end
+    it 'cannot enqueue the same job' do
+      expect { Resque.enqueue(described_class, *args) }.to_not change {
+        Resque.size(queue_name)
+      }.from(1)
+    end
 
-      it 'locks the mutex' do
-        expect { enqueue_job }.to change {
-          mutex.locked?
-        }.from(false).to(true)
-      end
+    it 'can execute the job' do
+      expect(execute_job).to_not be_nil
     end
   end
 
-  describe 'after dequeuing the job' do
-    let(:mutex) { described_class.mutex(args) }
-
-    subject(:dequeue_job) { Resque.dequeue(described_class, *args) }
+  context 'with one job in the queue and one job is being executed' do
+    before do
+      Resque.enqueue(described_class, *args)
+      expect(Resque.size(queue_name)).to eq(1)
+      worker_1.reserve
+      expect(Resque.size(queue_name)).to eq(0)
+      Resque.enqueue(described_class, *args)
+      expect(Resque.size(queue_name)).to eq(1)
+    end
 
-    before { mutex.lock! }
+    it 'cannot enqueue the same job' do
+      expect { Resque.enqueue(described_class, *args) }.to_not change {
+        Resque.size(queue_name)
+      }.from(1)
+    end
 
-    it 'unlocks the mutex' do
-      expect { dequeue_job }.to change {
-        mutex.locked?
-      }.from(true).to(false)
+    it 'can execute the job' do
+      expect(execute_job).to_not be_nil
+      expect { Resque.enqueue(described_class, *args) }.to change {
```

Comment:
Do you think this assertion is necessary? I think it's probably already covered by the 'with no jobs in the queue' context.